bigbite/themer
# [BBT-105] Display custom notification when related background and foreground colours fail accessible colour contrast ratios

## Description

[BBT-105](https://github.com/bigbite/themer/issues/105) - Based on the [colour contrast PR created here](https://github.com/bigbite/themer/pull/108), this PR adds a custom component to provide more visual feedback to the editor around the colour contrast of the `background` and `text` colour options.

In this component, the editor can see the selected colours contrast ratio and whether the colours comply to AA and AAA standards. This notice is placed above the `background` and `text` colour palettes.

## Change Log

- `src/editor/styles/index.scss` - Including styles
- `src/editor/components/StylesColor.js` - Adding custom component
- `src/editor/components/BBContrastChecker.js` - The custom contrast checker component
- `src/editor/styles/components/bb-contrast-checker.scss` - Adding styling for the custom component

## Steps to test

- Select a `block` or `element` that has the `text` and `background` colour options, such as `Elements > Button`
- Select a `text` colour from the available palette
- Select a _contrasting_ colour for the `background` from the palette
- Notice no warning is displayed
- Select the **same** colour for the `background` that is assigned to `text`
- Notice a contrast warning is now displayed

## Screenshots/Videos

[Custom colour contrast component demo](http://bigbite.im/v/aUlcaK)

## Checklist:
